Sources/SourceControl/RepositoryManager.swift

```diff
@@ -221,6 +239,11 @@ public class RepositoryManager {
                 DispatchQueue.global().async {
                     completion(result)
                 }
+                self.maxRequestsCondition.whileLocked {
+                    // We have processed this lookup request, decrement the request count and notify the blocked threads.
+                    self.requestCount -= 1
+                    self.maxRequestsCondition.broadcast()
```

Comment:
This can use `signal` not `broadcast` (since only one waiter will be able to pick up request).